### PR TITLE
CLN: Remove unused test code

### PR DESCRIPTION
**Pull request by simonjayhawkins**
cc @WillAyd same code as cleaned in #25647 

intention originally to just deal with the commented out instance of `pytest.raises` for #24332 but may have gone too far!

### CODE REVIEW

File: pandas/tests/computation/test_eval.py [hunk 1]
```diff
@@ -31,7 +31,6 @@
     assert_series_equal, makeCustomDataframe as mkdf, randbool)
 
 _series_frame_incompatible = _bool_ops_syms
```
Comment (simonjayhawkins) [inline pandas/tests/computation/test_eval.py]: `_series_frame_incompatible` can also be removed. AFK for a short while. will remove.

File: pandas/tests/computation/test_eval.py [hunk 2]
```diff
@@ -168,7 +167,15 @@ def test_complex_cmp_ops(self):
         for lhs, cmp1, rhs, binop, cmp2 in product(self.lhses, cmp_ops,
```
Comment (WillAyd) [inline pandas/tests/computation/test_eval.py]: Does it make sense to parametrize this?

**Review by WillAyd — COMMENTED** [review]
Great thanks as always for the PR. One comment on quick glance - will deep dive into review later

**Comment by simonjayhawkins** [comment]
I did start but got into a bit of a mess needs fixture isation first

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**